src/cpp/transport/tcp/RTCPMessageManager.cpp

```diff
@@ -382,16 +394,17 @@ TCPTransactionId RTCPMessageManager::sendLogicalPortIsClosedRequest(
     return sendLogicalPortIsClosedRequest(p_channel_resource, request);
 }
 
-TCPTransactionId RTCPMessageManager::sendUnbindConnectionRequest(TCPChannelResource *p_channel_resource)
+TCPTransactionId RTCPMessageManager::sendUnbindConnectionRequest(
+        std::shared_ptr<TCPChannelResource>& channel)
 {
     logInfo(RTCP_MSG, "Send [UNBIND_CONNECTION_REQUEST]");
     TCPTransactionId id = getTransactionId();
-    sendData(p_channel_resource, UNBIND_CONNECTION_REQUEST, id);
+    sendData(channel, UNBIND_CONNECTION_REQUEST, id);
     return id;
 }
 
 ResponseCode RTCPMessageManager::processBindConnectionRequest(
-        TCPChannelResource *p_channel_resource,
+        std::shared_ptr<TCPChannelResource>& channel,
```

Comment:
Using a reference instead of a copy can cause problems as the channel being freed while we are still using it.